src/Handlers/notifyCancellation/cancellation.notification.ts

```
@@ -0,0 +1,78 @@
+/* eslint-disable no-console */
+import { Request, Response } from "express";
+import axios from "axios";
+import { getToken } from "../../utils";
+import { slack } from "../../slack";
+import { logEvent } from "../../utils";
+
+const corelogicApiUrl = process.env.CORELOGIC_DIGITALHUB_API as string;
+const apiKey = process.env.CORELOGIC_DIGITALHUB_API_KEY as string;
+const apiCompanyId = process.env.CORELOGIC_DIGITALHUB_API_COMPANY_ID;
+
+export const notifyCancellation = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  let inspection,
+    inspectionId,
+    errorCode = 500;
+
+  try {
+    inspection = req.body.current;
+    inspectionId = inspection._id;
+    const externalId = inspection.meta?.external_id;
+
+    if (!externalId) {
+      errorCode = 400;
+      throw new Error(`Missing required field external_id`);
+    }
+
+    const token = await getToken();
+    const response = await axios.get(
+      `${corelogicApiUrl}/api/digitalhub/v1/Action/State?InspectionId=${externalId}&UniqueId=${inspectionId}&Action=Cancel`,
+      {
+        headers: {
+          Authorization: `Bearer ${token}`,
+          "Content-Type": "application/json",
+          "api-key": apiKey,
+          "api-companyid": apiCompanyId,
+        },
+      }
+    );
+
+    console.log(
+      `Successfully sent cancellation notification for inspection ${inspectionId}`
+    );
+
+    await logEvent({
+      inspection: inspectionId,
+      event: "sent_corelogic_inspection_cancellation_notification",
+      meta: { external_id: externalId },
+    });
+
+    res.status(200).send(response);
+  } catch (error) {
+    console.error(
+      `Error in sending cancellation notification for inspection ${inspectionId} of carrier ${inspection.carrier._id}`,
+      error
+    );
```

Should we throw an error in addition to console logging the error?

---

Besides console logging, an error message will be sent to slack and a 400/500 response will be returned. Re-throwing the error might simply result in a 500 response with the error, which the current code is already doing. Also, per google documentation, some types of uncaught exception could cause cold start and impact performance. We can discuss more.

---

Good to know, thanks for the info on that.